### ISSUE TITLE
Ridiculously large icons crash the application (2000x2000)

### ISSUE THREAD

**Issue by DDzwiedziu** [MultiMC/MultiMC5#2584]
System Information
-----------------------------
MultiMC version: 0.6.5-1217

Operating System: Ubuntu 18.04.2 LTS

Summary of the issue or suggestion:
----------------------------------------------
MultiMC crashes when I try to edit or run the instance of Enigmatica2 1.63a. With the icon from the Curse modpack page ( https://media.forgecdn.net/avatars/191/824/636861646865159125.png )

Happens both on a old and a fresh (only the `instances` directory is symlinked from the old one) instance of MultiMC

What should happen:
------------------------------
The "Edit Instance" window should appear or the instance should start.

Steps to reproduce the issue (Add more if needed):
-------------------------------------------------------------
1. Download the pack zip from https://minecraft.curseforge.com/projects/enigmatica2/files/2691579
2. Install the zip trough MultiMC new instance dialog.
3. Download the pack avatar from Curse (you need to use the element inspector).
4. Add the avatar as the pack icon.
5. Try to launch or edit the instance.

Suspected cause:
---------------------------
Unknown. My guesses would be on a mangled/malicious PNG or a library edge-case.

Logs/Screenshots:
----------------------------
[//]: # (Please refer to https://github.com/MultiMC/MultiMC5/wiki/Log-Upload for instructions on how to attach your logs.)
This is a log between clicking on "Edit Instance" and crashing:
```
    4.779 D Task "ComponentUpdateTask(0x2b77e30)" starting for the first time
    4.780 D Loading components
    4.780 D Task "NetJob(Download of meta file index.json)" starting for the first time
    4.780 D Remote loading is being run for metadata index
    4.781 D Task "NetJob(Download of meta file net.minecraft/1.12.2.json)" starting for the first time
    4.781 D Remote loading is being run for "net.minecraft"
    4.781 D Task "NetJob(Download of meta file net.minecraftforge/14.23.5.2781.json)" starting for the first time
    4.781 D Remote loading is being run for "net.minecraftforge"
    4.782 D Task "NetJob(Download of meta file net.minecraft/index.json)" starting for the first time
    5.264 C BAD stuff happened to mod json:
    5.264 C "{
  "modListVersion": "2",
  "modList":
  [
    {
      "modid": "openblocks",
      "name": "OpenBlocks",
      "description": "All the things you never thought you will ever need",
      "version": "1.8",
      "mcversion": "1.12.2",
      "url": "http://openmods.info/",
      "updateUrl": "",
      "authorList": [
        "Mikee",
        "NeverCast",
        "boq",
        "Lyqyd"
      ],
      "credits": "",
      "logoFile": "",
      "screenshots": [
      ],
      "parent": ""
    }
  ]
}
"
    5.475 W QMetaObject::connectSlotsByName: No matching signal for on_RunningState_changed(bool)
    5.496 D Applying "net.minecraft" GOOD
    5.496 D Applying "net.minecraftforge" ERROR
    5.584 W The X11 connection broke: Maximum allowed requested length exceeded (code 4)
XIO:  fatal IO error 2 (No such file or directory) on X server ":0"
      after 379 requests (379 known processed) with 0 events remaining.
```

Additional Info:
---------------------------

**Comment by lawremi** [comment]
Same problem here. Converting the image from png to jpg results in the same error. I noticed that the image was huge (over 2000x2000), so I scaled it down to 1024x1024, and it worked.

Two suggestions:
1. MultiMC should gracefully handle large images.
2. The modpack maintainer should scale down the icon. At 100% zoom it looks like it was scaled up anyway.

**Comment by peterix** [comment]
A 2000x2000 image used as an icon is ... insane.

**Comment by QuImUfu** [comment]
duplicate of #1663

**Comment by HellishINC** [comment]
I compiled your multimc-git package from the AUR and ran into this issue again. Icons were not downscaled as stated by pull #2695.
> fixes #2584, fixes #1663: Icons are now downscaled to 512x512 if they are bigger
